src/pqm/pqm.py

```diff
@@ -74,98 +427,94 @@ def _pqm_test(
     Returns
     -------
     tuple
-        Results from scipy.stats.chi2_contingency function.
+        Results from scipy.stats.chi2_contingency or the PyTorch implementation.
     """
-    nx, *D = x_samples.shape
-    ny, *D = y_samples.shape
+
+    # Determine if we're working with NumPy or PyTorch
+    is_numpy = isinstance(x_samples, np.ndarray) and isinstance(y_samples, np.ndarray)
+    is_torch = isinstance(x_samples, torch.Tensor) and isinstance(y_samples, torch.Tensor)
+    
+    if not (is_numpy or is_torch):
+        raise TypeError("x_samples and y_samples must both be either NumPy arrays or PyTorch tensors.")
+    
+    # Validate sample sizes
+    nx = x_samples.shape[0]
+    ny = y_samples.shape[0]
     if (nx + ny) <= num_refs + 2:
         raise ValueError(
-            "Number of reference samples (num_ref) must be less than the number of x/y samples. Ideally much less."
+            "Number of reference samples (num_refs) must be less than the number of x/y samples. Ideally much less."
         )
     elif (nx + ny) < 2 * num_refs:
         warnings.warn(
-            "Number of samples is small (less than twice the number of reference samples). Result will have high variance and/or be non-discriminating."
+            "Number of samples is small (less than twice the number of reference samples). "
+            "Result will have high variance and/or be non-discriminating."
         )
-    if whiten:
+    
+    # Z-score normalization
+    if z_score_norm:
         mean, std = _mean_std(x_samples, y_samples)
-        y_samples = (y_samples - mean) / std
-        x_samples = (x_samples - mean) / std
-
+        if is_numpy:
```

Comment:
You don't need the if statement since both numpy and torch versions are the same